src/backend/sgx/builder.rs

```diff
@@ -67,6 +67,20 @@ impl TryFrom<super::config::Config> for Builder {
             .ioctl(&mut file, &create)
             .context("Failed to create SGX enclave")?;
 
+        // TODO: Change to `perms::Unknown` after Linux kernel has a stable
+        // support for EMODPR.
+        std::mem::forget(unsafe {
+            Map::map(map.size())
+                .onto(map.addr())
+                .anonymously()
+                .known::<perms::ReadWriteExecute>(Kind::Shared)
+                .context(format!(
+                    "map error: {:#016x}-{:#016x}:RWX",
+                    map.addr(),
+                    map.addr() + map.size()
+                ))?
+        });
```

Comment:
We definitely don't want the whole enclave to be RWX...

Reply:
Yes, so how do I get the same location as where `ENARX_EXEC_END` is easily?

Reply:
And do you have any insight on the error I'm getting? It's good to remark that this patch on top does still give a full pass on tests. Only after last shim patch is applied the error messages start to appear.

Reply:
See this: https://github.com/enarx/enarx/pull/1616. Please remark that both drafts.

Reply:
```
test write_stderr ... ignored
Error: Segments overlap!
```
Do all mmaps go to implicitly to the segment table? Then this would be the trigger. I just wonder why it does not alread trigger w/o the shim change?

Reply:
Anyway, this I can fix. I'll update this PR shortly.